File: do/do.py
import serial
import threading

PORT = "COM39"
BAUD = 115200

ser = serial.Serial(PORT, BAUD, timeout=0.1)

def read_serial():
    while True:
        data = ser.readline()
        if data:
            print(data.decode(errors="replace").rstrip())

thread = threading.Thread(target=read_serial, daemon=True)
thread.start()

print("Serieller Monitor gestartet.")
print("Beispiele: r27,  y32,  g1,  o32,")
print("Beenden mit Strg+C")

while True:
    text = input("> ")
    ser.write((text + "\n").encode())

File: do/do.js
const { SerialPort } = require("serialport");
const readline = require("readline");

const PORT = "COM39";
const BAUDRATE = 115200;

const port = new SerialPort({
  path: PORT,
  baudRate: BAUDRATE,
});

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  prompt: "LED> ",
});

port.on("open", () => {
  console.log(`Verbunden mit ${PORT} @ ${BAUDRATE}`);
  console.log("Beispiele:");
  console.log("  r27  -> LED 27 rot");
  console.log("  y32  -> LED 32 gelb");
  console.log("  g1   -> LED 1 gruen");
  console.log("  o32  -> LED 32 aus");
  console.log("Beenden mit: exit");
  rl.prompt();
});

port.on("data", (data) => {
  process.stdout.write("\nArduino: " + data.toString());
  rl.prompt();
});

port.on("error", (err) => {
  console.error("SerialPort-Fehler:", err.message);
});

rl.on("line", (line) => {
  const command = line.trim();

  if (command === "exit" || command === "quit") {
    console.log("Beende...");
    rl.close();
    port.close();
    return;
  }

  if (command.length > 0) {
    port.write(command + "\n", (err) => {
      if (err) {
        console.error("Fehler beim Senden:", err.message);
      } else {
        console.log("Gesendet:", command);
      }
    });
  }

  rl.prompt();
});

rl.on("close", () => {
  if (port.isOpen) {
    port.close();
  }
  process.exit(0);
});
